### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by janeczku**
* Decouples setup routines from the init() funcs to prepare for unit testing
* Adds Init() method to providers interface
* Adds HealthCheck() method to providers interface for using least expensive API call
* Moves providers to individual packages
* Moves DNS types and helper funcs to utils package

### CODE REVIEW

**Comment by janeczku** [comment]
@alena1108 Please review

**Comment by janeczku** [comment]
https://github.com/rancher/rancher/issues/4596

File: dns/dns.go [hunk 1]
```diff
@@ -25,34 +17,13 @@ type ServiceDnsRecord struct {
...
 func ConvertToServiceDnsRecord(dnsRecord DnsRecord) ServiceDnsRecord {
 	splitted := strings.Split(dnsRecord.Fqdn, ".")
 	serviceRecord := ServiceDnsRecord{dnsRecord.Fqdn, splitted[0], splitted[1]}
 	return serviceRecord
 }
 
-func ConvertToFqdn(serviceName string, stackName string, environmentName string) string {
-	domainNameEntries := []string{serviceName, stackName, environmentName, RootDomainName}
-	return strings.ToLower(strings.Join(domainNameEntries, "."))
+func ConvertToFqdn(serviceName, stackName, environmentName, rootDomainName string) string {
```
Comment (alena1108) [inline dns/dns.go]: thinking we should rename this class to utils, so from now on all dns util functions go there. Thoughts?

**Comment by alena1108** [comment]
LGTM. Needs to be rebased + we would need a big catalog update PR :) as image, use major release number bumped up by 1

File: dns/dns.go (hunk 1 above)
Comment (janeczku) [inline dns/dns.go]: Makes sense.

**Comment by alena1108** [comment]
@janeczku could you rename that file, squash the commits and rebase the PR? I will merge it after that

**Comment by alena1108** [comment]
actually thinking whether we should merge 2 new external providers code in first? and rebase on top of it (and apply the same changes). Or merge this PR first, then make modifications for providers in their respective PRs. Up to you, let me know

**Comment by janeczku** [comment]
@alena1108 Rebased on top of the new providers.

**Pull request merged**